src/content/blog/why-not-effect.mdx

```diff
@@ -0,0 +1,43 @@
+
+I'm no stranger to the functional programming world: I used [Rambda](https://ramdajs.com/) in a large-scale project in the past, and [Purify](https://github.com/gigobyte/purify) more recently. I'm completely sold on the value of typed errors, and I think that `Either`, `Option`, and `Maybe` types are generally good ideas.
+
+So I dove into Effect for a few days, and we decided as a team not to use it. Here's why.
+
+## Documentation
+
+The biggest concern right at the top: documentation, the lack thereof. Effect has [narrative documentation](https://effect.website/docs/quickstart), as well as [API documentation](https://effect-ts.github.io/effect/docs/effect), but both are incredibly incomplete.
+
+I think in large part this is because the API surface – the number of functions and variables exported by Effect – is enormous. But regardless, there are just so many core APIs that are undocumented. Like take `Effect.gen`, the function that lets you run a series of effects using a generator and get the result at the end: [it has no JSDoc documentation at all](https://effect-ts.github.io/effect/effect/Either.ts.html#gen). The same for nearly all of the functional shorthand methods like [ap, product, and productMany](https://effect-ts.github.io/effect/effect/Option.ts.html#ap) on the Option and Effect types. I ran into undocumented APIs constantly when I was reading code examples and writing new code.
+
+The narrative documentation leaves a lot to be desired: in the extremely brief documentation about running effects, it recommends you use [Effect.runFork](https://effect.website/docs/guides/essentials/running-effects#runfork), but only provides a contrived example that doesn't show you how to do something like get the result of the computation, or an error, or really use the `runFork` method in practice.
+
+The level of documentation is the first reason why we aren't adopting Effect: I'm not onboarding new team members onto something with so little quality documentation available. And I'm afraid that the documentation won't get better, for two reasons: first, that Effect inherits a lot from the [fp-ts project](https://gcanti.github.io/fp-ts/modules/Either.ts.html), which has been around since 2017 but has a similarly low level of documentation. And second, the surface area of Effect seems to be only growing - there are [six ecosystem modules as well as Effect](https://effect.website/docs/other/api-reference) that are listed in the documentation, and then a lot of other unstable modules that are still in development, like `@effect/sql-drizzle`. We use [Drizzle](https://orm.drizzle.team/) as our ORM and probably would use `@effect/sql-drizzle`, but it's unstable and undocumented. And I fear that, if there's some incremental improvement to add documentation, it'll never catch up with also-increasing surface area to document.
```

Comment:
This paragraph also feels a bit harsh. Maybe add some positive note about what they could do if they wanted to improve?